`NoSuchElementException: No value present` when codegen goal is present but there is no wsdl2java configuration

When the `codegen` goal of quarkus-maven-plugin is present but there is no wsdl2java configuration, the following issue occurrs:

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:3.0.0.Final:generate-code (generate-code) on project camel-quarkus-integration-test-cxf-soap-ssl: Quarkus code generation phase has failed: InvocationTargetException: No value present -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.quarkus:quarkus-maven-plugin:3.0.0.Final:generate-code (generate-code) on project camel-quarkus-integration-test-cxf-soap-ssl: Quarkus code generation phase has failed
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:375)
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:351)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:215)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:171)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:163)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
        at org.mvndaemon.mvnd.builder.SmartBuilderImpl.buildProject(SmartBuilderImpl.java:190)
        at org.mvndaemon.mvnd.builder.SmartBuilderImpl$ProjectBuildTask.run(SmartBuilderImpl.java:208)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.apache.maven.plugin.MojoExecutionException: Quarkus code generation phase has failed
        at io.quarkus.maven.GenerateCodeMojo.generateCode(GenerateCodeMojo.java:85)
        at io.quarkus.maven.GenerateCodeMojo.doExecute(GenerateCodeMojo.java:53)
        at io.quarkus.maven.QuarkusBootstrapMojo.execute(QuarkusBootstrapMojo.java:154)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:370)
        ... 12 common frames omitted
Caused by: java.lang.reflect.InvocationTargetException: null
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.maven.GenerateCodeMojo.generateCode(GenerateCodeMojo.java:79)
        ... 16 common frames omitted
Caused by: java.util.NoSuchElementException: No value present
        at java.base/java.util.Optional.get(Optional.java:143)
        at io.quarkiverse.cxf.deployment.codegen.Wsdl2JavaCodeGen.scan(Wsdl2JavaCodeGen.java:122)
        at io.quarkiverse.cxf.deployment.codegen.Wsdl2JavaCodeGen.wsdl2java(Wsdl2JavaCodeGen.java:97)
        at io.quarkiverse.cxf.deployment.codegen.Wsdl2JavaCodeGen.trigger(Wsdl2JavaCodeGen.java:78)
        at io.quarkus.deployment.CodeGenerator.lambda$trigger$3(CodeGenerator.java:176)
        at io.quarkus.deployment.CodeGenerator.callWithClassloader(CodeGenerator.java:151)
        at io.quarkus.deployment.CodeGenerator.trigger(CodeGenerator.java:173)
        at io.quarkus.deployment.CodeGenerator.initAndRun(CodeGenerator.java:63)
        ... 21 common frames omitted
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -69,7 +69,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        testModule: [client, logging, mtom, mtom-awt, server, ws-addressing, ws-rm, ws-security-client, ws-security-server, ws-trust, wsdl2java]
+        testModule: [client, logging, mtom, mtom-awt, server, ws-addressing, ws-rm, ws-security-client, ws-security-server, ws-trust, wsdl2java, wsdl2java-no-config]
     name: ${{matrix.testModule}} native tests
     needs: build-and-run-jvm-tests
     runs-on: ubuntu-latest

--- a/extensions/core/deployment/src/main/java/io/quarkiverse/cxf/deployment/codegen/Wsdl2JavaCodeGen.java
+++ b/extensions/core/deployment/src/main/java/io/quarkiverse/cxf/deployment/codegen/Wsdl2JavaCodeGen.java
@@ -85,8 +85,9 @@ public class Wsdl2JavaCodeGen implements CodeGenProvider {
         }
 
         if (!result) {
-            log.warn("wsdl2java processed 0 WSDL files under "
-                    + absModuleRoot(context.inputDir()).relativize(context.inputDir()));
+            log.infof(
+                    "wsdl2java processed 0 WSDL files under %s",
+                    absModuleRoot(context.inputDir()).relativize(context.inputDir()));
         }
         return result;
     }
@@ -117,6 +118,10 @@ public class Wsdl2JavaCodeGen implements CodeGenProvider {
             String prefix,
             Map<String, String> processedFiles,
             Consumer<Path> wsdlFileConsumer) {
+
+        if (includes.isEmpty()) {
+            return false;
+        }
 
         final String selectors = "    " + prefix + ".includes = "
                 + includes.get().stream().collect(Collectors.joining(","))

--- a/integration-tests/pom.xml
+++ b/integration-tests/pom.xml
@@ -25,6 +25,7 @@
         <module>mtom</module>
         <module>mtom-awt</module>
         <module>wsdl2java</module>
+        <module>wsdl2java-no-config</module>
     </modules>
 
     <properties>

--- a/integration-tests/wsdl2java-no-config/README.adoc
+++ b/integration-tests/wsdl2java-no-config/README.adoc
@@ -1,0 +1,5 @@
+= `wsdl2java` code generation test
+
+The main focus of the current test module is to make sure that the `generate-code` mojo of
+`quarkus-maven-plugin` works well with `Wsdl2JavaCodeGen` even if there is no `wsdl2java` configuration
+in `application.properties`.

--- a/integration-tests/wsdl2java-no-config/pom.xml
+++ b/integration-tests/wsdl2java-no-config/pom.xml
@@ -1,0 +1,88 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>io.quarkiverse.cxf</groupId>
+        <artifactId>quarkus-cxf-integration-tests</artifactId>
+        <version>2.0.1-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>quarkus-cxf-integration-test-wsdl2java-no-config</artifactId>
+
+    <name>Quarkus CXF - Integration Test - wsdl2java no configuration</name>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkiverse.cxf</groupId>
+            <artifactId>quarkus-cxf</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkiverse.cxf</groupId>
+            <artifactId>quarkus-cxf-xjc-plugins</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>io.rest-assured</groupId>
+            <artifactId>rest-assured</artifactId>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>generate-code</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+
+    <profiles>
+        <profile>
+            <id>virtualDependencies</id>
+            <activation>
+                <property>
+                    <name>!noVirtualDependencies</name>
+                </property>
+            </activation>
+            <dependencies>
+                <!-- The following dependencies guarantee that this module is built after them. You can update them by running `mvn process-resources -Pformat -N` from the source tree root directory -->
+                <dependency>
+                    <groupId>io.quarkiverse.cxf</groupId>
+                    <artifactId>quarkus-cxf-deployment</artifactId>
+                    <version>${project.version}</version>
+                    <type>pom</type>
+                    <scope>test</scope>
+                    <exclusions>
+                        <exclusion>
+                            <groupId>*</groupId>
+                            <artifactId>*</artifactId>
+                        </exclusion>
+                    </exclusions>
+                </dependency>
+                <dependency>
+                    <groupId>io.quarkiverse.cxf</groupId>
+                    <artifactId>quarkus-cxf-xjc-plugins-deployment</artifactId>
+                    <version>${project.version}</version>
+                    <type>pom</type>
+                    <scope>test</scope>
+                    <exclusions>
+                        <exclusion>
+                            <groupId>*</groupId>
+                            <artifactId>*</artifactId>
+                        </exclusion>
+                    </exclusions>
+                </dependency>
+            </dependencies>
+        </profile>
+    </profiles>
+
+</project>

--- a/integration-tests/wsdl2java-no-config/src/main/resources/application.properties
+++ b/integration-tests/wsdl2java-no-config/src/main/resources/application.properties
@@ -1,0 +1,1 @@
+# Intentionally left empty - see README.adoc

--- a/integration-tests/wsdl2java-no-config/src/test/java/io/quarkiverse/cxf/wsdl2java/no/config/it/Wsdl2JavaNoConfigTest.java
+++ b/integration-tests/wsdl2java-no-config/src/test/java/io/quarkiverse/cxf/wsdl2java/no/config/it/Wsdl2JavaNoConfigTest.java
@@ -1,0 +1,22 @@
+package io.quarkiverse.cxf.wsdl2java.no.config.it;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class Wsdl2JavaNoConfigTest {
+
+    @Test
+    void codegenTests() throws IOException {
+
+        /* Make sure that no java files were generated */
+        final Path wsdl2java = Paths.get(
+                "target/generated-test-sources/wsdl2java");
+        Assertions.assertThat(wsdl2java).doesNotExist();
+
+    }
+
+}

--- a/integration-tests/wsdl2java-no-config/src/test/resources/wsdl/CalculatorService.wsdl
+++ b/integration-tests/wsdl2java-no-config/src/test/resources/wsdl/CalculatorService.wsdl
@@ -1,0 +1,230 @@
+<?xml version="1.0" ?><wsdl:definitions xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/" xmlns:tns="http://www.jboss.org/eap/quickstarts/wscalculator/Calculator" xmlns:soap="http://schemas.xmlsoap.org/wsdl/soap/" xmlns:ns1="http://schemas.xmlsoap.org/soap/http" name="CalculatorService" targetNamespace="http://www.jboss.org/eap/quickstarts/wscalculator/Calculator">
+  <wsdl:types>
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:tns="http://www.jboss.org/eap/quickstarts/wscalculator/Calculator" elementFormDefault="unqualified" targetNamespace="http://www.jboss.org/eap/quickstarts/wscalculator/Calculator" version="1.0">
+
+  <xs:element name="add" type="tns:add"></xs:element>
+
+  <xs:element name="addOperands" type="tns:addOperands"></xs:element>
+
+  <xs:element name="addOperandsResponse" type="tns:addOperandsResponse"></xs:element>
+
+  <xs:element name="addResponse" type="tns:addResponse"></xs:element>
+
+  <xs:element name="divide" type="tns:divide"></xs:element>
+
+  <xs:element name="divideResponse" type="tns:divideResponse"></xs:element>
+
+  <xs:element name="multiply" type="tns:multiply"></xs:element>
+
+  <xs:element name="multiplyResponse" type="tns:multiplyResponse"></xs:element>
+
+  <xs:element name="subtract" type="tns:subtract"></xs:element>
+
+  <xs:element name="subtractResponse" type="tns:subtractResponse"></xs:element>
+
+  <xs:complexType name="divide">
+    <xs:sequence>
+      <xs:element name="arg0" type="xs:int"></xs:element>
+      <xs:element name="arg1" type="xs:int"></xs:element>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="divideResponse">
+    <xs:sequence>
+      <xs:element name="return" type="xs:int"></xs:element>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="add">
+    <xs:sequence>
+      <xs:element name="arg0" type="xs:int"></xs:element>
+      <xs:element name="arg1" type="xs:int"></xs:element>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="addResponse">
+    <xs:sequence>
+      <xs:element name="return" type="xs:int"></xs:element>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="subtract">
+    <xs:sequence>
+      <xs:element name="arg0" type="xs:int"></xs:element>
+      <xs:element name="arg1" type="xs:int"></xs:element>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="subtractResponse">
+    <xs:sequence>
+      <xs:element name="return" type="xs:int"></xs:element>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="addOperands">
+    <xs:sequence>
+      <xs:element minOccurs="0" name="arg0" type="tns:operands"></xs:element>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="operands">
+    <xs:sequence>
+      <xs:element name="a" type="xs:int"></xs:element>
+      <xs:element name="b" type="xs:int"></xs:element>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="addOperandsResponse">
+    <xs:sequence>
+      <xs:element minOccurs="0" name="return" type="tns:result"></xs:element>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="result">
+    <xs:sequence>
+      <xs:element minOccurs="0" name="operands" type="tns:operands"></xs:element>
+      <xs:element name="result" type="xs:int"></xs:element>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="multiply">
+    <xs:sequence>
+      <xs:element name="arg0" type="xs:int"></xs:element>
+      <xs:element name="arg1" type="xs:int"></xs:element>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="multiplyResponse">
+    <xs:sequence>
+      <xs:element name="return" type="xs:int"></xs:element>
+    </xs:sequence>
+  </xs:complexType>
+
+</xs:schema>
+  </wsdl:types>
+  <wsdl:message name="add">
+    <wsdl:part element="tns:add" name="parameters">
+    </wsdl:part>
+  </wsdl:message>
+  <wsdl:message name="subtract">
+    <wsdl:part element="tns:subtract" name="parameters">
+    </wsdl:part>
+  </wsdl:message>
+  <wsdl:message name="subtractResponse">
+    <wsdl:part element="tns:subtractResponse" name="parameters">
+    </wsdl:part>
+  </wsdl:message>
+  <wsdl:message name="addResponse">
+    <wsdl:part element="tns:addResponse" name="parameters">
+    </wsdl:part>
+  </wsdl:message>
+  <wsdl:message name="multiplyResponse">
+    <wsdl:part element="tns:multiplyResponse" name="parameters">
+    </wsdl:part>
+  </wsdl:message>
+  <wsdl:message name="divideResponse">
+    <wsdl:part element="tns:divideResponse" name="parameters">
+    </wsdl:part>
+  </wsdl:message>
+  <wsdl:message name="divide">
+    <wsdl:part element="tns:divide" name="parameters">
+    </wsdl:part>
+  </wsdl:message>
+  <wsdl:message name="addOperandsResponse">
+    <wsdl:part element="tns:addOperandsResponse" name="parameters">
+    </wsdl:part>
+  </wsdl:message>
+  <wsdl:message name="addOperands">
+    <wsdl:part element="tns:addOperands" name="parameters">
+    </wsdl:part>
+  </wsdl:message>
+  <wsdl:message name="multiply">
+    <wsdl:part element="tns:multiply" name="parameters">
+    </wsdl:part>
+  </wsdl:message>
+  <wsdl:portType name="CalculatorService">
+    <wsdl:operation name="divide">
+      <wsdl:input message="tns:divide" name="divide">
+    </wsdl:input>
+      <wsdl:output message="tns:divideResponse" name="divideResponse">
+    </wsdl:output>
+    </wsdl:operation>
+    <wsdl:operation name="add">
+      <wsdl:input message="tns:add" name="add">
+    </wsdl:input>
+      <wsdl:output message="tns:addResponse" name="addResponse">
+    </wsdl:output>
+    </wsdl:operation>
+    <wsdl:operation name="subtract">
+      <wsdl:input message="tns:subtract" name="subtract">
+    </wsdl:input>
+      <wsdl:output message="tns:subtractResponse" name="subtractResponse">
+    </wsdl:output>
+    </wsdl:operation>
+    <wsdl:operation name="addOperands">
+      <wsdl:input message="tns:addOperands" name="addOperands">
+    </wsdl:input>
+      <wsdl:output message="tns:addOperandsResponse" name="addOperandsResponse">
+    </wsdl:output>
+    </wsdl:operation>
+    <wsdl:operation name="multiply">
+      <wsdl:input message="tns:multiply" name="multiply">
+    </wsdl:input>
+      <wsdl:output message="tns:multiplyResponse" name="multiplyResponse">
+    </wsdl:output>
+    </wsdl:operation>
+  </wsdl:portType>
+  <wsdl:binding name="CalculatorServiceSoapBinding" type="tns:CalculatorService">
+    <soap:binding style="document" transport="http://schemas.xmlsoap.org/soap/http"></soap:binding>
+    <wsdl:operation name="divide">
+      <soap:operation soapAction="" style="document"></soap:operation>
+      <wsdl:input name="divide">
+        <soap:body use="literal"></soap:body>
+      </wsdl:input>
+      <wsdl:output name="divideResponse">
+        <soap:body use="literal"></soap:body>
+      </wsdl:output>
+    </wsdl:operation>
+    <wsdl:operation name="add">
+      <soap:operation soapAction="" style="document"></soap:operation>
+      <wsdl:input name="add">
+        <soap:body use="literal"></soap:body>
+      </wsdl:input>
+      <wsdl:output name="addResponse">
+        <soap:body use="literal"></soap:body>
+      </wsdl:output>
+    </wsdl:operation>
+    <wsdl:operation name="subtract">
+      <soap:operation soapAction="" style="document"></soap:operation>
+      <wsdl:input name="subtract">
+        <soap:body use="literal"></soap:body>
+      </wsdl:input>
+      <wsdl:output name="subtractResponse">
+        <soap:body use="literal"></soap:body>
+      </wsdl:output>
+    </wsdl:operation>
+    <wsdl:operation name="addOperands">
+      <soap:operation soapAction="" style="document"></soap:operation>
+      <wsdl:input name="addOperands">
+        <soap:body use="literal"></soap:body>
+      </wsdl:input>
+      <wsdl:output name="addOperandsResponse">
+        <soap:body use="literal"></soap:body>
+      </wsdl:output>
+    </wsdl:operation>
+    <wsdl:operation name="multiply">
+      <soap:operation soapAction="" style="document"></soap:operation>
+      <wsdl:input name="multiply">
+        <soap:body use="literal"></soap:body>
+      </wsdl:input>
+      <wsdl:output name="multiplyResponse">
+        <soap:body use="literal"></soap:body>
+      </wsdl:output>
+    </wsdl:operation>
+  </wsdl:binding>
+  <wsdl:service name="CalculatorService">
+    <wsdl:port binding="tns:CalculatorServiceSoapBinding" name="Calculator">
+      <soap:address location="http://172.17.0.3:8080/calculator-ws/CalculatorService"></soap:address>
+    </wsdl:port>
+  </wsdl:service>
+</wsdl:definitions>

